src/openrct2/world/Map.cpp

```diff
@@ -1345,7 +1348,6 @@ void MapUpdateTiles()
         }
 
         gGrassSceneryTileLoopPosition++;
-        gGrassSceneryTileLoopPosition &= 0xFFFF;
```

Comment:
Was this supposed to be the "MapIsEdge"

Reply:
The the type is uint16_t and its used to continuously update the grass, from that variable it extracts the tile position. Given its uint16_t that operation makes no sense.

Reply:
The algorithm is a bit funky. It tries to update grass with a arbitary scan, so to distribute the growing over the whole map. This was tied to the 256x256 map (even if the map was smaller).

When increasing the map size limit, I opted to do the same 256x256 algorithm repeatedly across the whole map. So the whole map is divided into 256x256 chunks - the algorithm works like before on each chunk per update.

Reply:
I noticed that, I wonder if we should maybe just the the prng to select the next tile to update, but all of this I would prefer to do in another PR. As for the line that was removed, this doesn't affect the logic.

Reply:
Each tile needs to be updated before each one is updated again. So for rng to work, it would need to shuffle a list of all the tiles which isn't as memory efficient.

Reply:
I'm thinking more about a simple random distribution which would seem a bit more natural in my opinion, the prng is not too bad and has a decent distribution and grass just grows wherever it wants to grow, but I don't think we should discuss this any further here.

Reply:
A uniform distribution will look more natural than the current grass update algorithm. The whole reason I immediately recognized the pattern in Matt's gif that led to this fix is pretty much proof of that.